http/src/handler.rs

```diff
@@ -311,13 +343,57 @@ impl<M: Metadata, S: Middleware<M>> RpcHandler<M, S> {
 			Method::Options => {
 				RpcHandlerState::Writing(Response::empty())
 			},
+			// Respond to health API request if there is one configured.
+			Method::Get if self.health_api.as_ref().map(|x| &*x.0) == Some(request.uri().path()) => {
+				RpcHandlerState::ProcessHealth {
+					metadata,
+					method: self.health_api.as_ref()
+							.map(|x| x.1.clone())
+							.expect("Health api is defined since the URI matched."),
+				}
+			},
 			// Disallow other methods.
 			_ => {
 				RpcHandlerState::Writing(Response::method_not_allowed())
 			},
 		}
 	}
 
+	fn process_health(
+		&self,
+		method: String,
+		metadata: M,
+	) -> Result<RpcPollState<M, S::Future>, hyper::Error> {
+		use self::core::types::{Call, MethodCall, Version, Params, Request, Id, Output, Success, Failure};
+
+		// Parse request
```

Comment:
what exactly is parsed here

Reply:
Good catch, blidndly copied a comment from `process_rest` (which was incorrect as well). Fixed.